### PR TITLE
Add .gitattributes

### PR DESCRIPTION
**Pull request by Zerorigin**
Add .gitattributes to avoid Encoding and EOL formatting changes.

### CODE REVIEW

**Review by isaak654 — CHANGES_REQUESTED** [review]
~1. I think the crlf line ending should not be set for all .bat and .cmd files, but only for specific .bat and .cmd files that might present [this issue](https://serverfault.com/a/795638) in future such as [RunReport.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/msgs/RunReport.bat), [build.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/build.bat), [ParseVersion.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ParseVersion.bat), [configure.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/SandboxiePlus/QtSingleApp/configure.bat) and so on, if you can find other similarities.~ -> https://github.com/sandboxie-plus/Sandboxie/pull/2188/commits/8c1292f898d193b5eedcc2672ee98647614e5442
2. I didn't check them all, but in case make sure to verify that all .bat and .cmd files are already in UTF-8.

**Comment by Zerorigin** [comment]
> 1. I think the crlf line ending should not be set for all .bat and .cmd files, but only for specific .bat and .cmd files that might present [this issue](https://serverfault.com/a/795638) in future such as [RunReport.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/msgs/RunReport.bat), [build.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/build.bat), [ParseVersion.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ParseVersion.bat), [configure.bat](https://github.com/sandboxie-plus/Sandboxie/blob/master/SandboxiePlus/QtSingleApp/configure.bat) and so on, if you can find other similarities.
> 2. I didn't check them all, but in case make sure to verify that all .bat and .cmd files are already in UTF-8.

https://stackoverflow.com/questions/11303405/force-encode-from-us-ascii-to-utf-8-iconv
https://www.shellhacks.com/find-out-text-file-line-endings-lf-or-clrf/

```
# file $(find . | grep -E "*.(bat|cmd)$")
./Installer/copy_build.cmd:                ASCII text
./Installer/get_openssl.cmd:               ASCII text
./Installer/get_qttranslations.cmd:        ASCII text
./Installer/merge_builds.cmd:              DOS batch file, ASCII text
./Sandboxie/install/build.bat:             DOS batch file, ASCII text
./Sandboxie/install/ParseVersion.bat:      ASCII text
./Sandboxie/msgs/RunReport.bat:            ASCII text
./Sandboxie/msgs/SbieText.bat:             ASCII text
./SandboxiePlus/install_jom.cmd:           ASCII text
./SandboxiePlus/qmake_plus.cmd:            ASCII text
./SandboxiePlus/QtSingleApp/configure.bat: ASCII text
./TestCI.cmd:                              ASCII text


# enca -d -L none $(find . | grep -E "*.(bat|cmd)$")
./Installer/copy_build.cmd: 7bit ASCII characters
./Installer/get_openssl.cmd: 7bit ASCII characters
./Installer/get_qttranslations.cmd: 7bit ASCII characters
./Installer/merge_builds.cmd: 7bit ASCII characters
./Sandboxie/install/build.bat: 7bit ASCII characters
./Sandboxie/install/ParseVersion.bat: 7bit ASCII characters
./Sandboxie/msgs/RunReport.bat: 7bit ASCII characters
./Sandboxie/msgs/SbieText.bat: 7bit ASCII characters
./SandboxiePlus/install_jom.cmd: 7bit ASCII characters
./SandboxiePlus/qmake_plus.cmd: 7bit ASCII characters
./SandboxiePlus/QtSingleApp/configure.bat: 7bit ASCII characters
./TestCI.cmd: 7bit ASCII characters


# enca -m -L none $(find . | grep -E "*.(bat|cmd)$")
./Installer/copy_build.cmd: US-ASCII
./Installer/get_openssl.cmd: US-ASCII
./Installer/get_qttranslations.cmd: US-ASCII
./Installer/merge_builds.cmd: US-ASCII
./Sandboxie/install/build.bat: US-ASCII
./Sandboxie/install/ParseVersion.bat: US-ASCII
./Sandboxie/msgs/RunReport.bat: US-ASCII
./Sandboxie/msgs/SbieText.bat: US-ASCII
./SandboxiePlus/install_jom.cmd: US-ASCII
./SandboxiePlus/qmake_plus.cmd: US-ASCII
./SandboxiePlus/QtSingleApp/configure.bat: US-ASCII
./TestCI.cmd: US-ASCII
```

It seems all .bat and .cmd files are the crlf line ending and use US-ASCII also call 7bit ASCII or UTF-8?

**Comment by isaak654** [comment]
> 1. At first I simply followed here. 

I’ve changed my mind since then, so I just pushed https://github.com/sandboxie-plus/Sandboxie/commit/8827ecea214c3935e99a68a4009bc493ca36063d to reflect that.

**Comment by Zerorigin** [comment]
I checked them all again.
I just pushed the new commit to reflect the changes.

```
for i in $(find . | grep -E "*.(bat|cmd)$")
do
    grep -E "(call|goto)" -l $i
done


./Installer/merge_builds.cmd
./Sandboxie/install/build.bat
./Sandboxie/install/ParseVersion.bat
./Sandboxie/msgs/RunReport.bat
./SandboxiePlus/qmake_plus.cmd
./SandboxiePlus/QtSingleApp/configure.bat
```
```
for i in $(find . | grep -E "*.(bat|cmd)$")
do
    export bytes_to_scan=$(wc -c < $i)
    file -P bytes=$bytes_to_scan $i
done


./Installer/copy_build.cmd: ASCII text, with CRLF line terminators
./Installer/get_openssl.cmd: ASCII text, with CRLF line terminators
./Installer/get_qttranslations.cmd: ASCII text, with CRLF line terminators
./Installer/merge_builds.cmd: DOS batch file, ASCII text, with CRLF line terminators
./Sandboxie/install/build.bat: DOS batch file, ASCII text, with CRLF line terminators
./Sandboxie/install/ParseVersion.bat: ASCII text, with CRLF line terminators
./Sandboxie/msgs/RunReport.bat: ASCII text, with CRLF line terminators
./Sandboxie/msgs/SbieText.bat: ASCII text, with CRLF line terminators
./SandboxiePlus/install_jom.cmd: ASCII text, with CRLF line terminators
./SandboxiePlus/qmake_plus.cmd: ASCII text, with CRLF line terminators
./SandboxiePlus/QtSingleApp/configure.bat: ASCII text, with CRLF line terminators
./TestCI.cmd: ASCII text, with CRLF line terminators
```

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_core_autocrlf
+
+[attr]binary -diff -merge -text
+*.bat text working-tree-encoding=UTF-8
+*.cmd text working-tree-encoding=UTF-8
+*.ts text working-tree-encoding=UTF-8 eol=lf
+*.md text working-tree-encoding=UTF-8 eol=lf
+Installer/isl/*.isl text eol=lf
+
+# Set crlf EOL for *.bat and *.cmd files that contain call/goto block.
```
Comment (isaak654) [inline .gitattributes:12]:
```suggestion
# Set crlf EOL for *.bat and *.cmd files to prevent a label parser issue with lf.
```

**Review by isaak654 — CHANGES_REQUESTED** [review]

**Comment by isaak654** [comment]
🤔  Both [main.yml](https://github.com/sandboxie-plus/Sandboxie/blob/master/.github/workflows/main.yml) and [lupdate.yml](https://github.com/sandboxie-plus/Sandboxie/blob/master/.github/workflows/lupdate.yml) are failing with some of the changes:
CI
lupdate

**Comment by isaak654** [comment]
A few workarounds / tools to consider:
https://stackoverflow.com/a/46151520
https://unix.stackexchange.com/a/381263
https://github.com/archiecobbs/bom
https://github.com/jlblancoc/utf-bom-utils

**Review by isaak654 — APPROVED** [review]

**Pull request merged**